### PR TITLE
[Docs] Adds section on installing AMIs.

### PR DESCRIPTION
**Pull request by tcsc**

### CODE REVIEW

File: docs/pages/installation.mdx [hunk 1]
```diff
@@ -122,6 +122,25 @@ We provide pre-built `amd64`, `arm`, and `arm64` Docker images for every version
 </TabItem>
 </Tabs>
 
+## Amazon EC2
```
Comment (stevenGravy) [inline docs/pages/installation.mdx:125]: WE should reference these are meant to be used with the supported AWS Terraform and link to those instructions.  Listing all on their own could give the impression you can just use the AMIs directly w/o it.  it's not designed for that AFAIK.

File: docs/pages/installation.mdx [hunk 2]
```diff
@@ -122,6 +122,25 @@ We provide pre-built `amd64`, `arm`, and `arm64` Docker images for every version
...
+
+We also provide pre-built `amd64` Amazon Linux 2023 based EC2 AMIs with Teleport pre-installed.
```
Comment (stevenGravy) [inline docs/pages/installation.mdx:127]:
```suggestion
We provide pre-built `amd64` Amazon Linux 2023 based EC2 AMIs with Teleport pre-installed.
```

File: docs/pages/installation.mdx (hunk 1 above)
Comment (ptgott) [reply docs/pages/installation.mdx:125]: If our AMIs are only supposed to be used with our AWS Terraform module, I'm not sure if we should put AMI instructions in the installation guide.  This is one of our most visited guides, and a frequent starting point for new users, and we're intending for organizations to self-host a production Teleport cluster only if they have a certain size or specific needs.

Comment (r0mant) [reply docs/pages/installation.mdx:125]: It should be possible to use AMIs standalone but users would need to provide a few [template variables](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/starter-cluster/data.tpl) for config generation.

I think it's ok to keep this on the installation page (increases visibility into hardened AMIs as well, like Paul mentioned), but we should mention these template variables and give a link to our terraform guides too saying that they can use these with terraform. WDYGT?

Comment (stevenGravy) [reply docs/pages/installation.mdx:125]: I would make part of that a include so it can be resused in the AWS terraform guide.

Comment (ptgott) [reply docs/pages/installation.mdx:125]: Mentioning these on the installation page sounds good as long as we:
- Mention the template variables (like Roman said)
- Indicate how the AMI works in terms of Teleport configuration and cluster architecture, i.e., can a user deploy agents via the AMI? Is it an Auth/Proxy single-instance deployment? Can it work with the HA setup we recommend in [this guide](https://goteleport.com/docs/deploy-a-cluster/high-availability/)?

Comment (ptgott) [reply docs/pages/installation.mdx:335]: @tcsc What do you think about the feedback above? I'm still not sure from this section of the Installation page how or when you would use these AMIs.

Comment (tcsc) [reply docs/pages/installation.mdx:335]: Updating this right now to address te above. Will ping you when its ready for another pass.

**Review by ptgott — APPROVED** [review]

**Comment by ptgott** [comment]
@tcsc Added backport labels. Please feel free to remove any that are inappropriate!

**Review by r0mant — APPROVED** [review]

**Pull request merged**